File: CryptoScan.Subscriptions.API/Models/Subscription.cs
using System.ComponentModel.DataAnnotations;
using MongoDB.Bson;
using MongoDB.Bson.Serialization.Attributes;

namespace CryptoScan.Subscriptions.API.Models;

public record Subscription
{
  [BsonId]
  [BsonRepresentation(BsonType.ObjectId)]
  [StringLength(24, MinimumLength = 24)]
  public string? SubscriptionId { get; init; }

  public string UserId { get; init; }
  
  public string Email { get; init; }

  public CryptocurrencySymbol Symbol { get; init; }

  public double? Threshold { get; init; }
  
  public double? PercentageThreshold { get; init; }

  public TimeRange TimeRange { get; init; }

  public Trend Trend { get; init; } = Trend.Unspecified;
}

File: CryptoScan.Monitoring.Service/src/main/java/com/cryptoscan/monitoring/model/subscription/Subscription.java
package com.cryptoscan.monitoring.model.subscription;

import lombok.Getter;
import lombok.Setter;
import lombok.ToString;

@Getter
@Setter
@ToString
public class Subscription {
    private String SubscriptionId;
    private String UserId;
    private String Email;
    private CryptocurrencySymbol Symbol;
    private Double threshold;
    private Double percentageThreshold;
    private TimeRange timeRange;
    private Trend trend;
}
